torchmetrics/metric.py

```diff
@@ -413,30 +412,36 @@ def __setstate__(self, state: Dict[str, Any]) -> None:
         self.update: Callable = self._wrap_update(self.update)  # type: ignore
         self.compute: Callable = self._wrap_compute(self.compute)  # type: ignore
 
-    @property
-    def dtype(self) -> "torch.dtype":
-        """Return the dtype of the metric."""
-        return self._dtype
-
-    @dtype.setter
-    def dtype(self, new_dtype: "torch.dtype") -> None:
-        # necessary to avoid infinite recursion
-        raise RuntimeError("Cannot set the dtype explicitly. Please use module.to(new_dtype).")
-
     @property
     def device(self) -> "torch.device":
         """Return the device of the metric."""
         return self._device
 
     def to(self, *args: Any, **kwargs: Any) -> "Metric":
-        """Moves and/or casts the parameters and buffers.
+        """Moves the parameters and buffers.
 
-        Works similar to nn.Module.to but also updates the metrics device and dtype properties
+        Normal dtype casting is not supported by this method instead use the `set_dtype` method instead.
         """
-        # there is diff nb vars in PT 1.5
         out = torch._C._nn._parse_to(*args, **kwargs)
-        self._update_properties(device=out[0], dtype=out[1])
-        return super().to(*args, **kwargs)
+        if len(out) == 4:  # pytorch 1.5 and higher
+            device, dtype, non_blocking, convert_to_format = out
```

Comment:
what's the minimum version of pytorch torchmetrics supports?

Reply:
pytorch v1.3